src/main/java/org/sunbird/workflow/service/impl/SignUpWFServiceImpl.java

```diff
@@ -0,0 +1,240 @@
+                    applicationStatus = new WfStatusEntity();
+                    wfId = UUID.randomUUID().toString();
+                    applicationStatus.setWfId(wfId);
+                    applicationStatus.setServiceName(wfRequest.getServiceName());
+                    applicationStatus.setUserId(wfRequest.getUserId());
+                    applicationStatus.setApplicationId(wfRequest.getApplicationId());
+                    applicationStatus.setRootOrg(rootOrg);
+                    applicationStatus.setOrg(org);
+                    applicationStatus.setCreatedOn(new Date());
+                    wfRequest.setWfId(wfId);
+                }
+                applicationStatus.setLastUpdatedOn(new Date());
+                applicationStatus.setCurrentStatus(nextState);
+                applicationStatus.setActorUUID(wfRequest.getActorUserId());
+                applicationStatus.setUpdateFieldValues(mapper.writeValueAsString(wfRequest.getUpdateFieldValues()));
+                applicationStatus.setInWorkflow(!wfNextStatus.getIsLastState());
+                applicationStatus.setDeptName(wfRequest.getDeptName());
+                wfStatusRepo.save(applicationStatus);
+                if (wfStatus.getNotificationEnable() == Boolean.TRUE) {
+                    producer.push(configuration.getWorkFlowNotificationTopic(), wfRequest);
+                }
+                producer.push(configuration.getWorkflowApplicationTopic(), wfRequest);
+            }
+        } catch (IOException e) {
+            errMsg = Constants.WORKFLOW_PARSING_ERROR_MESSAGE;
+            log.error(errMsg);
+        }
+        response.getParams().setErrmsg(errMsg);
+        HashMap<String, Object> data = new HashMap<>();
+        data.put(Constants.WF_ID_CONSTANT, wfId);
+        data.put(Constants.STATUS, nextState);
+        data.put(Constants.MESSAGE, Constants.STATUS_CHANGE_MESSAGE + nextState);
+        response.getResult().putAll(data);
+        return response;
+    }
+
+    private String validateUserAndWfStatus(WfRequest wfRequest, WfStatus wfStatus, WfStatusEntity applicationStatus) {
+        List<String> params = new ArrayList<String>();
+        StringBuilder strBuilder = new StringBuilder();
+        if (StringUtils.isEmpty(wfRequest.getWfId()) && !wfStatus.getStartState()) {
+            params.add(Constants.WORKFLOW_ID_ERROR_MESSAGE);
+        }
+
+        if ((!ObjectUtils.isEmpty(applicationStatus))
+                && (!wfRequest.getState().equalsIgnoreCase(applicationStatus.getCurrentStatus()))) {
+            params.add("Application is in " + applicationStatus.getCurrentStatus()
+                    + " State but trying to be move in " + wfRequest.getState() + " state!");
+        }
+        if (!params.isEmpty()) {
+            strBuilder.append(params);
+        }
+        return strBuilder.toString();
+    }
+
+    public SBApiResponse getWfApplication(String rootOrg, String org, String wfId, String applicationId) {
```

Comment:
Don't pass multiple arguments.Create a class and pass an object.For other methods also